src/core_plugins/timelion/public/panels/timechart/schema.js

```diff
@@ -168,15 +168,15 @@ module.exports = function timechartFn(Private, config, $rootScope, timefilter, $
 
         let legendScope = $scope.$new();
         function drawPlot(plotConfig) {
+          if (!$('.chart-canvas', $elem).length) $elem.html(template);
+          const canvasElem = $('.chart-canvas', $elem);
 
-          if (!plotConfig || !plotConfig.length) {
-            $elem.empty();
+          // we can't use `$.plot` to draw the chart when the height or width is 0
+          // so, we'll need another event to trigger drawPlot to actually draw it
+          if (canvasElem.height() === 0 || canvasElem.width() === 0) {
             return;
```

Comment:
That's the interesting part, the setTimeout was calling drawPlot with an empty `plotConfig` so the redraw logic actually ended up just calling `$elem.empty()`. Another event was coming through and calling `drawPlot` which actually caused it to draw correctly.